src/com/facebook/buck/jvm/kotlin/KotlinLibraryDescription.java

```diff
@@ -193,8 +193,38 @@ public void findDepsForTargetFromConstructorArgs(
   }
 
   public interface CoreArg extends JavaLibraryDescription.CoreArg {
+    /**
+     * @deprecated use {@link #getFreeCompilerArgs()}.
+     */
+    @Deprecated
     ImmutableList<String> getExtraKotlincArguments();
 
+    ImmutableList<String> getFreeCompilerArgs();
+
+    Optional<Boolean> getAllWarningsAsErrors();
```

Comment:
```suggestion
    @Value.Default
    default boolean getAllWarningsAsErrors() {
      return false;
    }    
```

you can provide defaults for all booleans

Reply:
Does `@Value.Default` ensure that an implementation still gets generated under the hood too?

Reply:
yes, it does create an implementation which fallbacks to the default implementation if not set.

Reply:
02d8940